sklearn/kernel_ridge.py

```diff
@@ -0,0 +1,185 @@
+"""The :mod:`sklearn.kernel_ridge` module implements kernel ridge regression."""
+
+# Authors: Mathieu Blondel <mathieu@mblondel.org>
+#          Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+# License: BSD 3 clause
+
+import warnings
+
+import numpy as np
+from scipy import linalg
+
+from .utils import check_X_y
+from sklearn.base import BaseEstimator, RegressorMixin
+from sklearn.metrics.pairwise import pairwise_kernels
+from sklearn.linear_model.ridge import _solve_cholesky_kernel
+
+
+class KernelRidge(BaseEstimator, RegressorMixin):
+    """Kernelized ridge regression.
+
+    Kernelized ridge regression (KRR) combines ridge regression (linear least
+    squares plus l2-norm regularization) with the kernel trick. It thus
+    learns a linear function in the space induced by the respective kernel and
+    the data. For non-linear kernels, this corresponds to a non-linear
+    function in the original space.
+
+    The model learned by KRR is identical to support vector regression (SVR).
+    However, different loss functions are used (ridge versus  epsilon-
+    insensitive loss). In contrast to SVR, fitting a KRR can be done in closed-
+    form and is typically faster for medium-sized datasets. On the other hand,
+    the learned model is non-sparse and thus slower than SVR at prediction-time.
+
+    Using KRR with a linear kernel can be advisable over non-kernelized Ridge
+    in situations when the number of feature dimensions D is considerably larger
```

Comment:
Nitpick: I think this is preferably represented by `P` instead of `D`. Same thing three lines lower